### PR TITLE
options/posix: Stub several functions

### PR DESCRIPTION
**Pull request by Dennisbonke**
Part of the mlibc LFS project.

### CODE REVIEW

**Comment by ArsenArsen** [comment]
Implement these instead. The only acceptable stubs here are `gethostby*_r`, and even those should be implemented by refactoring their non-reentrant counterparts to just use a global buffer and the reentrant ones (IIRC, we implement these?)

File: options/posix/include/unistd.h [hunk 1]
```diff
@@ -252,6 +252,16 @@ int pipe2(int *pipefd, int flags);
...
 
+int getresuid(uid_t *ruid, uid_t *euid, uid_t *suid);
+int getresgid(gid_t *rgid, gid_t *egid, gid_t *sgid);
+
+// Glibc extensions
+int getdomainname(char *name, size_t len);
+int setdomainname(const char *name, size_t len);
+
+#define ftruncate64 ftruncate
+#define truncate64 truncate
```
Comment (avdgrinten) [inline options/posix/include/unistd.h:263]: Unrelated change.

Comment (Dennisbonke) [reply options/posix/include/unistd.h:263]: Whoops, my bad. I'll remove them.

**Review by Geertiebear — COMMENTED** [review]
LGTM but the CI is failing on this PR.

**Review by Geertiebear — APPROVED** [review]
LGTM, but there are merge conflicts.
